server/website/views/content_objects.py

```diff
@@ -0,0 +1,373 @@
+
+            elif (db_type == "page"):
+                self.ser = ContentPageSerializer(data=self.initial_data)
+
+            elif (db_type == "list"):
+                items = self.initial_data.pop("items")
+                self.ser = ContentCollectionListSerializer(data=self.initial_data, items=items)
+
+            elif (db_type == "dict"):
+                items = self.initial_data.pop("items")
+                self.ser = ContentCollectionSerializer(data=self.initial_data, items=items)
+
+
+
+    def is_valid(self, raise_exception=False):
+        """Checks if the data of the serializer is valid"""
+        assert hasattr(self, 'initial_data'), (
+            'Cannot call `.is_valid()` as no `data=` keyword argument was '
+            'passed when instantiating the serializer instance.'
+        )
+        if not hasattr(self, '_validated_data'):
+            if hasattr(self, 'ser'):
+                if self.ser.is_valid():
+                    self._validated_data = self.initial_data
+                    self._errors = {}
+
+                else:
+                    self._errors = self.ser.errors
+            else:
+                self._errors = {"db_type":_("Invalid db_type")}
+
+        if self._errors and raise_exception:
+            raise ValidationError(self.errors)
+
+        return not bool(self._errors)
+
+    def save(self, **kwargs):
+        """
+        If the data is valid, saves the ContentObject to the database
+        """
+        assert hasattr(self, '_errors'), (
+            'You must call `.is_valid()` before calling `.save()`.'
+        )
+
+        assert not self.errors, (
+            'You cannot call `.save()` on a serializer with invalid data.'
+        )
+
+        # Guard against incorrect use of `serializer.save(commit=False)`
+        assert 'commit' not in kwargs, (
+            "'commit' is not a valid keyword argument to the 'save()' method. "
+            "If you need to access data before committing to the database then "
+            "inspect 'serializer.validated_data' instead. "
+            "You can also pass additional keyword arguments to 'save()' if you "
+            "need to set extra attributes on the saved model instance. "
+            "For example: 'serializer.save(owner=request.user)'.'"
+        )
+
+        assert not hasattr(self, '_data'), (
+            "You cannot call `.save()` after accessing `serializer.data`."
+            "If you need to access data before committing to the database then "
+            "inspect 'serializer.validated_data' instead. "
+        )
+
+
+        return self.ser.save(**kwargs)
+
+
+class ContentObjectView(generics.CreateAPIView):
+    """Create view for all types of content objects"""
+    serializer_class = ContentObjectBaseSerializer
+    queryset = ContentObjectBase.objects.all()
+
+    def post(self, request, *args, **kwargs):
+        request.data["containing_page"] = kwargs["containing_page_pk"]
```

Comment:
Får ett Attributerror på den här raden med meddelandet "This QueryDict instance is immutable" tror man måste kopiera den?